### PR TITLE
loginfirstcommit

### PR DESCRIPTION
**Pull request by AravindKaay**
Login Functionality is working

### CODE REVIEW

File: src/components/button.tsx [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+export type ButtonComponentPropTypes = ButtonProps & {
+  label: string | ReactNode;
+  suffix?: ReactNode;
+  prefixStyle?: CSSProperties;
+  suffixStyle?: CSSProperties;
+  onPrefixClick?: MouseEventHandler;
+  onSuffixClick?: MouseEventHandler;
+  wrapperStyle?: CSSProperties;
+  loading?: boolean;
+};
```
Comment (MJyotheesh) [inline src/components/button.tsx:13]: give one line space after types

File: src/components/button.tsx [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+        <span className="icon-label">{label}</span>
+      )}
+      {suffix && (
+        <span style={suffixStyle} onClick={onSuffixClick}>
+          {suffix}
+        </span>
+      )}
+    </Button>
+  );
+};
```
Comment (MJyotheesh) [inline src/components/button.tsx:54]: here too

File: src/components/index.tsx [hunk 3]
```diff
@@ -1,4 +1,6 @@
 import { SideMenu } from "./side-menu";
 import { Wrapper } from "./wrapper";
+import InputComponent from "./input"
+import ButtonComponent from "./button"
 
-export { Wrapper, SideMenu };
\ No newline at end of file
+export { Wrapper, SideMenu,InputComponent,ButtonComponent };
```
Comment (MJyotheesh) [inline src/components/index.tsx:6]: create generics folder in components then move button and input components there

File: src/components/input.tsx [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+    name,
+    required = true,
+    placeholder,
+    wrapperStyle,
+    error,
+    suffix = false,
+    textStyle,
+    labelStyle,
+    inputStyle,
+    ...prop
```
Comment (MJyotheesh) [inline src/components/input.tsx:35]: what is prop?

File: src/components/input.tsx [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+            autoComplete="off"
+          />
+          {suffix ? (
+            <div className="suffix" onClick={() => setShowInput(!showInput)}>
+              {showInput ? <EyeCross Height={20} Width={20} /> : <EyeIcon />}
+            </div>
+          ) : null}
+        </InputGroup>
+      </div>
+      {error && <div className="error_div">{error}</div>}
```
Comment (MJyotheesh) [inline src/components/input.tsx:73]: don't use underscore in class  names

File: src/components/styles/button.scss [hunk 6]
```diff
@@ -0,0 +1,12 @@
...
+}
+
+.icon-label {
+    padding: 10px 10px;
+}
+
+.spinner-border {
+    max-height: 21px;
+    max-width: 21px;
+}
```
Comment (MJyotheesh) [inline src/components/styles/button.scss:12]: styles also create one folder called generics and move these files to there

File: src/pages/Home/index.tsx [hunk 7]
```diff
@@ -0,0 +1,9 @@
+import React from 'react'
+
+const Home = () => {
+  return (
+    <div>Home coming</div>
```
Comment (MJyotheesh) [inline src/pages/Home/index.tsx:5]: haha

File: src/shared/api/url-helper.ts [hunk 8]
```diff
@@ -1,3 +1,4 @@
 import http from './http';
 
 export const getDriversList = (params: object) => http.get('driver/get-all', params);
```
Comment (MJyotheesh) [inline src/shared/api/url-helper.ts:3]: remove the above api

File: src/shared/assets/Mask Group 2.png
Comment (MJyotheesh) [inline src/shared/assets/Mask Group 2.png:1]: change the file name without spaces

File: src/shared/constant-values.ts [hunk 9]
```diff
@@ -1 +1,14 @@
...
+export const whatAppNumber = '1234567890';
+
+export const RegexpList = {
+    password: /^(?=.*[0-9])(?=.*[!@#$%^&*])[a-zA-Z0-9!@#$%^&*]{8,13}$/,
+    phone: /^\d{10}$/,
+    email: /^[a-zA-Z0-9.!#$%&'*+/=?^_`{|}~-]+@[a-zA-Z0-9-]+(?:\.[a-zA-Z0-9-]+)*$/
+};
+
+export const mobileScreen = 'https://fleetmatch-images.s3.amazonaws.com/mobile-screen.png'
+export const desktopScreen = 'https://fleetmatch-images.s3.amazonaws.com/desktop-screen.png';
```
Comment (MJyotheesh) [inline src/shared/constant-values.ts:14]: except regexplist and userRole other than that anything useful here

File: src/shared/formValidation/index.ts [hunk 10]
```diff
@@ -0,0 +1,38 @@
...
+                errors[field] = invalid;
+        } else {
+            errors[field] =
+                typeof required === "string" ? required : `Please Enter the ${field}`;
+        }
+    }
+    return { errors };
+};
+
+export default Validate;
```
Comment (MJyotheesh) [inline src/shared/formValidation/index.ts:38]: don't copy directly use only needful things

File: src/pages/Login/index.tsx [hunk 11]
```diff
@@ -0,0 +1,118 @@
...
+    const { errors } = Validate(schema, formData);
+    setError(errors);
+    if (Object.keys(errors).length) return setLoading(false);
+
+    try {
+      const { userName, password } = formData;
+      const userData: any = await loginAdmin({
+        userName,
+        password: password,
+        role: userRole,
```
Comment (MJyotheesh) [inline src/pages/Login/index.tsx:46]: no need to send role it will work

File: src/pages/Login/index.tsx
Comment (MJyotheesh) [inline src/pages/Login/index.tsx:1]: remove unwanted empty lines

File: src/pages/Login/index.tsx [hunk 12]
```diff
@@ -0,0 +1,118 @@
...
+      });
+      setSessionStorage("user", { userName, token: userData.result });
+      navigate("/");
+    } catch (err: any) {
+      const error = await err.json();
+      console.log("catch")
+      const { data = "Something Went Wrong Please Try Again" } = error;
```
Comment (MJyotheesh) [inline src/pages/Login/index.tsx:53]: move to constant file like services

File: src/pages/Login/index.tsx [hunk 13]
```diff
@@ -0,0 +1,118 @@
...
+      setLoading(false);
+      alert(data);
```
Comment (MJyotheesh) [inline src/pages/Login/index.tsx:55]: implement notification refer supplier web

**Review by MJyotheesh — CHANGES_REQUESTED** [review]
Fix the comments

File: public/index.html [hunk 14]
```diff
@@ -5,8 +5,8 @@
     <meta charset="UTF-8" />
     <meta http-equiv="X-UA-Compatible" content="IE=edge" />
     <meta name="viewport" content="width=device-width, initial-scale=1.0" />
-    <script src="https://cdn.jsdelivr.net/npm/bootstrap@5.3.0/dist/js/bootstrap.bundle.min.js"></script>
-    <link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/bootstrap@5.3.0/dist/css/bootstrap.min.css" integrity="sha384-Gn5384xqQ1aoWXA+058RXPxPg6fy4IWvTNh0E263XmFcJlSAwiGgFAW/dAiS6JXm" crossorigin="anonymous">
+    <script src="https://cdn.jsdelivr.net/npm/bootstrap@5.2.3/dist/js/bootstrap.bundle.min.js"></script>
```
[the remark this answers is not in the document]
Comment (AravindKaay) [reply public/index.html:8]: The earlier link code is not working . I replaced it by copying from bootstrap documentation.

File: src/components/styles/variables.scss [hunk 15]
```diff
@@ -0,0 +1,25 @@
...
+    --grey-shade-8: #707070;
+    
+    --font-family-1: Montserrat;
+ 
+    --blue-shade-1: #005EB8;
+ 
+    --black-shade-1: #000;
+    --black-shade-2: #1D1D1D;
+ 
+    --brown-shade-1: #222222;
```
[the remark this answers is not in the document]
Comment (AravindKaay) [reply src/components/styles/variables.scss:23]: To differentiate between different color groups as u could see black color group is separated from brown group with a space which improves readability and it doesn't look messy.

**Review by MJyotheesh — APPROVED** [review]

**Pull request merged**